app/acme.py

```diff
@@ -213,18 +230,18 @@ def final(self, keynum, csr):
             "kid": self.kid,
         }
         payload = {
-            #    "csr": b64encode(csr).decode().replace('+','-').replace('/','_'),
-            "csr": urlsafe_b64encode(csr)
-            .decode()
-            .rstrip("="),
+            "csr": urlsafe_b64encode(csr).decode().rstrip("="),
         }
         self.debugrequest(protected, payload)
         token = jwt.JWS(payload=json.dumps(payload))
         token.add_signature(self.key, alg="ES256", protected=protected)
-        headers = {"Content-Type": "application/jose+json"}
-        response = requests.post(
-            self.finalize[keynum], data=token.serialize(), headers=headers, timeout=60
-        )
+        headers = {
+            "Content-Type": "application/jose+json",
+            "X-Acme-Jwt": jwt_token,
```

Comment:
Normally this is already sent as a challenge and we should not need to update the finalize request.